File: classes-de-um-jogo.js

// classe e dentro dela uma função com propriedades
class hero{
    constructor(name, age, type) {
        this.name = name;
        this.age = age;
        this.type = type;
    }
// dando comportamento para classe com metodos que é uma função(dentro de classe não precisa utilizar a palavra function)
attack() {
    let ataque = '';

    switch (this.type) {
        case 'mago':
            ataque = 'magia';
            break;
        case 'guerreiro':
            ataque = ' espada';
            break;
        case 'monge':
            ataque = ' artes marciais';
            break;
        case 'ninja':
            ataque = 'shuriken';
            break;
        default:
            ataque = ' um ataque desconhecido';
    }

    console.log(`O ${this.type} atacou usando ${ataque}`);
}
}
// objetos que usa a classe instancia a classe
let paladino = new hero("paladino", 20, "guerreiro");
let arcanoLendario = new hero("arcanoLendario", 25, "mago");
let monge = new hero("monge", 30, "monge");
let ninjago = new hero("ninjago", 26, "ninja");

//chamar os métodos nas instâncias da classe 
/* O método attack() verifica o tipo do herói e atribui o ataque apropriado com base nesse tipo. Quando você chama o método attack() em cada instância do herói, ele imprimirá a mensagem correta de acordo com o tipo.*/

paladino.attack();
arcanoLendario.attack();
monge.attack();
ninjago.attack();

File: classes-de-um-jogo.py
## Ele cria instâncias da classe Hero e chama o método attack() em cada instância
## self é uma convenção que se refere ao próprio objeto da classe em um método.

class Hero:
    def __init__(self, name, age, type):
        self.name = name
        self.age = age
        self.type = type
 # Dentro do método, podemos acessar os atributos da instância usando self
    def attack(self):
        ataque = ''

        if self.type == 'mago':
            ataque = 'usou magia'
        elif self.type == 'guerreiro':
            ataque = 'usou espada'
        elif self.type == 'monge':
            ataque = 'usou artes marciais'
        elif self.type == 'ninja':
            ataque = 'usou shuriken'
        else:
            ataque = 'usou um ataque desconhecido'

        print(f'O {self.type} atacou usando {ataque}')
 # Criando uma instância da classe Hero
paladino = Hero("paladino", 20, "guerreiro")
arcanoLendario = Hero("arcanoLendario", 25, "mago")
monge = Hero("monge", 30, "monge")
ninjago = Hero("ninjago", 26, "ninja")
# Chamando o método attack na instância
paladino.attack()
arcanoLendario.attack()
monge.attack()
ninjago.attack()
